google/cloud/functions/http_request.h

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FUNCTIONS_FRAMEWORK_CPP_GOOGLE_CLOUD_FUNCTIONS_HTTP_REQUEST_H
+#define FUNCTIONS_FRAMEWORK_CPP_GOOGLE_CLOUD_FUNCTIONS_HTTP_REQUEST_H
+
+#include "google/cloud/functions/version.h"
+#include <map>
+#include <memory>
+#include <string>
+
+namespace google::cloud::functions {
+inline namespace FUNCTIONS_FRAMEWORK_CPP_NS {
+
+/**
+ * Represents a HTTP request.
+ *
+ * Functions to handle HTTP requests receive an object of this class.
+ */
+class HttpRequest {
+ public:
+  using Headers = std::multimap<std::string, std::string>;
```

Comment:
To make it clear that this is a type, consider naming `HeaderType` (similar to `value_type`, `size_type`, etc.)

Reply:
Done.